src/core_plugins/metrics/public/components/data_format_picker.js

```diff
@@ -17,25 +30,94 @@ class DataFormatPicker extends Component {
   handleChange(value) {
     if (value.value === 'custom') {
       this.handleCustomChange();
+    } else if (value.value === 'duration') {
+      const { from, to, decimals } = this.state;
+      this.props.onChange({
+        value: `${from},${to},${decimals}`
+      });
     } else {
       this.props.onChange(value);
     }
   }
 
+  handleDurationChange(name) {
+    return (value) => {
+      if (name === 'decimals') {
+        value = this.decimals;
+      }
+      this.setState({
+        [name]: value.value
+      }, () => {
+        const { from, to, decimals } = this.state;
+        this.props.onChange({
+          value: `${from},${to},${decimals}`
+        });
+      });
+    };
+  }
+
   render() {
```

Comment:
there's somewhat of a weird issue where the `duration` option becomes unavailable.

Steps:
- just use default chart (default Count metric)
- in options, select the duration Data Formatter
- change some settings (e.g format from weeks to milliseconds)
- remove the decimal places entry so the textbox is empty -> the form bounces back to the `custom` data formatter.
- try and go back to the Duration Formatter -> it won't let you, always bouncing back to the default `custom` data formatter